### PR TITLE
Add build recipe to actually compile singularity

### PR DESCRIPTION
**Pull request by eherot**

### CODE REVIEW

File: cookbook/recipes/build.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  user       node[:singularity][:user]
+  action     :export
+end
+
+execute 'build_singularity' do
+  action  :run
+  # Maven (or rather npm) has issues with
+  # being run as root.
+  user    node[:singularity][:user]
+  environment('HOME' => '/home/singularity')
```
Comment (wolfd) [inline cookbook/recipes/build.rb]: Is this supposed to be hardcoded?

Comment (eherot) [inline cookbook/recipes/build.rb]: Good catch. Will fix.

**Comment by tpetr** [comment]
LGTM, but I'd only suggest using this installation method (building straight from git) for testing / playing with Singularity.

The preferred installation method for production environments (in my opinion) would be to download a pre-compiled released JAR from Maven (i.e. the `shaded.jar` links on http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.hubspot%22%20AND%20a%3A%22SingularityService%22). Could we add support for this too?

**Comment by eherot** [comment]
I would be totally in favor of adding that. I didn't even know it existed until just now.

**Pull request merged**